hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1276,8 +1276,18 @@ public function upgrade_1029() {
    * Makes Hour Location id field autonumeric and adds id as a primary key.
    */
   public function upgrade_1030() {
-    CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` ADD PRIMARY KEY (`id`)');
-    CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` CHANGE `id` `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT');
+    try {
+      CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` ADD PRIMARY KEY (`id`)');
+      CRM_Core_DAO::executeQuery('ALTER TABLE `civicrm_hrhours_location` CHANGE `id` `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT');
+    } catch (Exception $e) {
+      $tableStructure = CRM_Core_DAO::executeQuery('DESC civicrm_hrhours_location');
+
+      while ($tableStructure->fetch()) {
+        if ($tableStructure->Field == 'id' && ($tableStructure->Key != 'PRI' || stripos($tableStructure->Extra, 'auto_increment') === false)) {
```

Comment:
can't you use the exception message to tell what was the cause of the error instead of inspecting the table description?

If no, then please make this more readable. Assign the whole condition to a variable that describes what is being checked and use this variable inside the if